src/base.jl

```diff
@@ -536,6 +536,42 @@ function get_units_base(system::System)
     return string(system.units_settings.unit_system)
 end
 
+"""
+`with_units_base(function, system, units::Union{UnitSystem, String}; suppress_units_info = true)`
+
+A "context manager" similar to [`Logging.with_logger`](@ref) that sets the system's units
+base to the given value, executes the function, then sets the units base back. If
+`suppress_units_info`, suppresses logging below `Warn` from internal calls made to
+facilitate this. Not thread safe.
+
+# Examples
+```julia
+active_power_mw = with_units_base(sys, UnitSystem.NATURAL_UNITS) do
+    get_active_power(gen)
+end
+# now active_power_mw is in natural units no matter what units base the system is in
+```
+"""
+function with_units_base(f::Function, sys::System, units::Union{UnitSystem, String};
+    suppress_units_info = true)
+    units_logger = if suppress_units_info
+        x -> Logging.with_logger(x, Logging.SimpleLogger(Logging.Warn))
```

Comment:
Something like this is needed to prevent getting spammed by `[ Info: Unit System changed to...` messages, but the current implementation has the undesired consequence of suspending any fancy logging the user has configured, directing any warning and error logs from `set_units_base_system!` (of which, granted, there should be none) only to the console. @daniel-thom do you know of a simple way to temporarily filter out all logs below a given level without replacing existing loggers? `Logging.disable_logging` is almost there, but there doesn't seem to be a public way to query the state to return to…. If not, maybe we change the interface to "suppresses _all_ logging from internal calls….

Reply:
Or maybe I've overcomplicated this and should just bypass `set_units_base_system!` altogether.

Reply:
If I understood your last point, you would have a separate code path that does not issue the log message, right? If so, I like that.

Reply:
Done